### DISCUSSION TITLE
OpenTelemetry Client Libraries

### DISCUSSION THREAD

**Discussion by bvinayakumar** [jaegertracing/jaeger#2885] [Q&A]
Hello,

As per Jaeger documentation, OpenTelemetry (https://www.jaegertracing.io/docs/1.22/opentelemetry/) is currently in experimental state.

Are there any supported Jaeger client libraries available (or under development) for OpenTelemetry that I can experiment with? 
I am particularly looking for Java and Node.js client libraries.

Kind regards,
Vinaya

**Comment by albertteoh** [comment]
With OpenTelemetry, you would use the OpenTelemetry client libraries to emit OTLP spans to [OTEL collector](https://github.com/open-telemetry/opentelemetry-collector) which can then export into Jaeger:

**java**: https://github.com/open-telemetry/opentelemetry-java
**node.js**: https://github.com/open-telemetry/opentelemetry-js

It seems you can emit Jaeger spans too ([java](https://github.com/open-telemetry/opentelemetry-java/blob/2ba472d0e5f56c635d30fa945aeba06c6dd25bed/exporters/jaeger/README.md), [node.js](https://github.com/open-telemetry/opentelemetry-js/blob/1eed360e4e151854a7feaf253350e510201e6dcc/packages/opentelemetry-exporter-jaeger/README.md)) which you could send straight to Jaeger's collector.

**Comment by jpkrohling** [comment]
That page is about the experimental *backend* components that use OpenTelemetry Collector as the base.  As @albertteoh mentioned, the OpenTelemetry client libraries can be used to send spans both to Jaeger directly, as it's a requirement stated in the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md#requirements
